### PR TITLE
Add `IgnoreDslMethods` option to `Style/MethodCallWithArgsParentheses`

### PR DESCRIPTION
**Pull request by Drenmi**
This cop was almost unusable in its current state, registering hundreds of false positives in a relatively simple application. 😅 

This change introduces a `IgnoreDslMethods` option (enabled by default) that ignores method calls that:

 - don't have a receiver, and
 - whose parent is a class- or module body.

-----------------

Before submitting the PR make sure the following are checked:

* [X] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [X] Used the same coding conventions as the rest of the project.
* [X] Feature branch is up-to-date with `master` (if not - rebase it).
* [X] Squashed related commits together.
* [x] Added tests.
* [X] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [X] All tests are passing.
* [X] The new code doesn't generate RuboCop offenses.
* [X] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [X] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by bbatsov** [comment]
> This change introduces a IgnoreDslMethods option (enabled by default) that ignores method calls that:

I think such methods are typically called `macro style` methods in Ruby lingo. But I can imagine some different cases like `it` and so on.

**Comment by Drenmi** [comment]
@bbatsov: I agree. You have a good point. Since nested blocks are not covered, naming it DSL will likely be confusing. Changed to macro and rebased. 😀

**Comment by bbatsov** [comment]
You forgot to change the commit message and the changelog.

File: config/default.yml [hunk 1]
```diff
@@ -743,6 +743,7 @@ Style/LambdaCall:
     - braces
 
 Style/MethodCallWithArgsParentheses:
+  IgnoreDslMethods: true
```
Comment (bbatsov) [inline config/default.yml]: I think this doesn't belong to this commit.

File: spec/rubocop/cop/style/method_call_with_args_parentheses_spec.rb [hunk 2]
```diff
@@ -80,4 +80,34 @@
     inspect_source(cop, 'puts :test')
     expect(cop.offenses).to be_empty
   end
+
+  context 'with DSL like methods' do
```
Comment (bbatsov) [inline spec/rubocop/cop/style/method_call_with_args_parentheses_spec.rb]: This as well.

**Comment by Drenmi** [comment]
@bbatsov: Fixed. 😅

**Comment by bbatsov** [comment]
👍
